### ISSUE TITLE
Allow specification of dependencies

### ISSUE THREAD

**Issue by vreuter** [epigen/pypiper#58]
It's nice to not have `pypiper` be dependency-free, but that allows installation and then potential use of `ngstk`, for example, without required dependencies since many are Imported at the function level. Allow installation based on need with something like `pip install --upgrade pypiper[ngstk]`

**Comment by nsheff** [comment]
probably, ngstk should just be a separate package...

**Comment by nsheff** [comment]
(in the long run)

**Comment by vreuter** [comment]
Yeah definitely. Until then, the implementation for this allows a la carte installation to prevent import errors.
